src/hb-ot-math.cc

```diff
@@ -87,6 +87,20 @@ hb_position_t
 hb_ot_math_get_constant (hb_font_t *font,
 			 hb_ot_math_constant_t constant)
 {
+  /* https://github.com/harfbuzz/harfbuzz/issues/4653
+   * Cambria Math has incorrect value for displayOperatorMinHeight, and
+   * apparently Microsoft implementation swaps displayOperatorMinHeight and
+   * delimitedSubFormulaMinHeight, so we do the same if we detect Cambria Math
+   * with the swapped values. */
+  if ((constant == HB_OT_MATH_CONSTANT_DISPLAY_OPERATOR_MIN_HEIGHT ||
+       constant == HB_OT_MATH_CONSTANT_DELIMITED_SUB_FORMULA_MIN_HEIGHT) &&
+      font->face->table.MATH->is_bad_cambria (font))
+  {
+    if (constant == HB_OT_MATH_CONSTANT_DISPLAY_OPERATOR_MIN_HEIGHT)
+      constant = HB_OT_MATH_CONSTANT_DELIMITED_SUB_FORMULA_MIN_HEIGHT;
+    else
+      constant = HB_OT_MATH_CONSTANT_DISPLAY_OPERATOR_MIN_HEIGHT;
+  }
```

Comment:
Not sure how easy that will be, but I wonder if we could have a test for that in test/api/test-ot-layout-math.c

Reply:
A test will need a font with a MATH table that is exactly the same size as Cambria’s, but short of copying Cambria’s MATH table, I’m nit sure how to make such a font.

Reply:
I think creating a font with the same constants is easy. About the table_blob->length, I'm not quite sure, I guess you can try and add dummy entries in the table but not sure you can get exactly the same length.

Reply:
I’ve added a test, by padding MATH table with zeros to be exactly the expected length.